library/src/main/java/com/getstream/sdk/chat/utils/MessageListItemLiveData.java

```diff
@@ -135,60 +147,17 @@ public void observe(@NonNull LifecycleOwner owner,
             Log.i(TAG, "broadcast because reads changed");
             broadcastValue();
         });
-        this.messages.observe(owner, messages -> {
-            if (messages == null || messages.size() == 0) return;
-            // update based on messages
-            hasNewMessages = false;
-            String newlastMessageID = messages.get(messages.size() -1).getId();
-            if (!newlastMessageID.equals(lastMessageID)) {
-                hasNewMessages = true;
-            }
-            lastMessageID = newlastMessageID;
-            List<MessageListItem> entities = new ArrayList<MessageListItem>();
-            // iterate over messages and stick in the date entities
-            Message previousMessage = null;
-            int size = messages.size();
-            int topIndex = Math.max(0, size -1);
-            for (int i = 0; i < size; i++) {
-                Message message = messages.get(i);
-                Message nextMessage = null;
-                if (i +1 <= topIndex){
-                    nextMessage = messages.get(i+1);
-                }
 
-                // determine if the message is written by the current user
-                Boolean mine = message.getUser().equals(currentUser);
-                // determine the position (top, middle, bottom)
-                User user = message.getUser();
-                List<MessageViewHolderFactory.Position> positions = new ArrayList<MessageViewHolderFactory.Position>();
-                if (previousMessage == null || !previousMessage.getUser().equals(user)) {
-                    positions.add(MessageViewHolderFactory.Position.TOP);
-                }
-
-                if (nextMessage == null || !nextMessage.getUser().equals(user)) {
-                    positions.add(MessageViewHolderFactory.Position.BOTTOM);
-                }
-
-                if (previousMessage != null && nextMessage != null) {
-                    if (previousMessage.getUser().equals(user) && nextMessage.getUser().equals(user)) {
-                        positions.add(MessageViewHolderFactory.Position.MIDDLE);
-                    }
-                }
-                // date separator
-                if (previousMessage != null && !isSameDay(previousMessage, message)) {
-                    entities.add(new MessageListItem(message.getCreatedAt()));
-                }
+        messages.observe(owner, messages->{
+            if (threadMessages.getValue() != null) return;
```

Comment:
what happens when a new message is added to the channel while you have the thread open? it looks like it will ignore those messages.